include/phasar/DataFlow/IfdsIde/Solver/FlowFunctionCache.h

```diff
@@ -0,0 +1,353 @@
+  decltype(auto) getRetFlowFunction(ProblemTy &Problem,
+                                    ByConstRef<n_t> CallSite,
+                                    ByConstRef<f_t> Callee,
+                                    ByConstRef<n_t> ExitInst,
+                                    ByConstRef<n_t> RetSite, uint64_t CSExitId,
+                                    uint64_t CalleeRSId) {
+    if constexpr (needs_cache_v<ret_ff_t>) {
+      if constexpr (std::is_base_of_v<
+                        llvm::Instruction,
+                        std::remove_cv_t<std::remove_pointer_t<n_t>>>) {
+        if (llvm::isa<llvm::CallInst>(CallSite)) {
+          auto &Ret = SimpleRetFFCache.getOrCreate(CSExitId);
+          if (!Ret) {
+            Ret =
+                Problem.getRetFlowFunction(CallSite, Callee, ExitInst, RetSite);
+          }
+          return wrapFlowFunction(Problem, Ret);
+        }
+      }
+
+      auto &Ret = RetFFCache.getOrCreate(std::make_pair(CSExitId, CalleeRSId));
+      if (!Ret) {
+        Ret = Problem.getRetFlowFunction(CallSite, Callee, ExitInst, RetSite);
+      }
+      return wrapFlowFunction(Problem, Ret);
+    } else {
+      return wrapFlowFunction(
+          Problem,
+          Problem.getRetFlowFunction(CallSite, Callee, ExitInst, RetSite));
+    }
+  }
+
+  decltype(auto) getSummaryFlowFunction(ProblemTy &Problem,
+                                        ByConstRef<n_t> CallSite,
+                                        ByConstRef<f_t> Callee,
+                                        uint64_t /*CSCalleeId*/) {
+    if constexpr (needs_cache_v<summary_ff_t>) {
+      return GenericFlowFunction<d_t>(
+          Problem.getSummaryFlowFunction(CallSite, Callee));
+    } else {
+      return Problem.getSummaryFlowFunction(CallSite, Callee);
+    }
+
+    /// XXX: The old FECache doesn't cache summary FFs, so refrain from doing
+    /// this here as well. Enable it, once the user-problems reliably return
+    /// std::nullptr_t such that caching can be disabled automatically
+
+    // return getFlowFunction<summary_ff_t>(
+    //     SummaryFFCache, CSCalleeId, Problem,
+    //     [CallSite, Callee](ProblemTy &Problem) {
+    //       return Problem.getSummaryFlowFunction(CallSite, Callee);
+    //     });
+  }
+
+  void clear() noexcept {
+    NormalFFCache.clear();
+    CallFFCache.clear();
+    RetFFCache.clear();
+    SimpleRetFFCache.clear();
+    CtrFFCache.clear();
+    SummaryFFCache.clear();
+  }
+
+  void reserve(size_t NumInsts, size_t NumCalls, size_t NumFuns) {
+    detail::IntraFlowFunctionsMixin<normal_ff_t, ctr_ff_t>::reserveIntraFFs(
+        NumInsts, NumCalls);
+    CallFFCache.reserve(NumCalls);
+    RetFFCache.reserve(NumFuns);
+    SimpleRetFFCache.reserve(NumFuns);
+  }
+
+  void dumpStats(llvm::raw_ostream &OS) const { OS << getStats(); }
+
+  [[nodiscard]] FlowFunctionCacheStats getStats() const noexcept {
+    /// TODO: With C++20 use designated aggregate initializers here!
```

Comment:
```suggestion
    /// TODO (issue-ref): With C++20 use designated aggregate initializers here!
```
If you want to keep this in the code base, maybe add at least a tracking issue.

Reply:
done